src/mjolnir/ingest_transit.cc

```diff
@@ -474,18 +477,18 @@ bool write_stop_pair(
 
       // add information from calendar.txt and calendar_dates.txt
       auto* service_dow = stop_pair->mutable_service_days_of_week();
-      service_dow->Add(trip_calendar->monday == gtfs::CalendarAvailability::Available);
-      service_dow->Add(trip_calendar->tuesday == gtfs::CalendarAvailability::Available);
-      service_dow->Add(trip_calendar->wednesday == gtfs::CalendarAvailability::Available);
-      service_dow->Add(trip_calendar->thursday == gtfs::CalendarAvailability::Available);
-      service_dow->Add(trip_calendar->friday == gtfs::CalendarAvailability::Available);
-      service_dow->Add(trip_calendar->saturday == gtfs::CalendarAvailability::Available);
-      service_dow->Add(trip_calendar->sunday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(gtfs::Monday & dow_mask);
+      service_dow->Add(gtfs::Tuesday & dow_mask);
+      service_dow->Add(gtfs::Wednesday & dow_mask);
+      service_dow->Add(gtfs::Thursday & dow_mask);
+      service_dow->Add(gtfs::Friday & dow_mask);
+      service_dow->Add(gtfs::Saturday & dow_mask);
+      service_dow->Add(gtfs::Sunday & dow_mask);
 
       bool had_added_date = false;
-      for (const auto& cal_date_item : trip_calDates) {
-        auto d = to_local_pivot_sec(cal_date_item.date.get_raw_date());
-        if (cal_date_item.exception_type == gtfs::CalendarDateException::Added) {
+      for (auto cal_itr = trip_calDates.first; cal_itr != trip_calDates.second; ++cal_itr) {
```

Comment:
i had a bug here in the first draft of this pr where i didnt make a copy of the first iterator in the range and instead modified the original. this mean that for other stop pairs that use teh same trip they wouldnt get these exceptions which causes the tests to fail, so thanks to the tests this was caught!